bin/oconf

```diff
@@ -1,80 +1,52 @@
 #!/usr/bin/env node
-var oconf = require('../lib/index');
-
-var argp = require('optimist')
-    .usage('Usage: $0 [--lint|--extract-option <option> [--allow-missing-option][--option-as-json]] <file>')
-    .option('help', {
-        alias: 'h',
-        description: 'This help text.',
-        type: 'boolean'
-    })
-    .option('ignore', {
-        alias: 'i',
-        description: 'Absolute path or glob pattern for ignoring included files.',
-        type: 'string'
-    })
-    .option('lint', {
-        alias: 'l',
-        description: 'Lint the input file and report any errors.',
-        type: 'boolean'
-    })
-    .option('relaxed', {
-        alias: 'r',
-        description: 'Use relaxed JSON format.',
-        type: 'boolean'
-    })
-    .option('extract-option', {
-        alias: 'o',
-        description: 'Extract the value of an option.',
-        type: 'string'
-	})
-    .option('option-as-json', {
-        alias: 'j',
-        description: 'The extracted value will be extracted as JSON.',
-        type: 'boolean'
-    })
-    .option('public', {
-        alias: 'p',
-        description: 'Only values in #public objects will be extracted.',
-        type: 'boolean'
-    })
-    .option('json', { type: 'boolean' })
-    .option('allow-missing-option', {
-        alias: 'm',
-        description: 'Do not fail when no such value is present.',
-        type: 'boolean'
-    })
-    .option('allowmissing', { type: 'boolean' });
-
-var argv = argp.argv;
+const oconf = require('../lib/index');
+const { Command } = require('commander');
+const program = new Command();
+
+program
+    .name('oconf')
+    .usage('[--lint|--extract-option <option> [--allow-missing-option][--option-as-json]] <file>')
+    .option('-h, --help', 'This help text.')
+    .option('-i, --ignore <path>', 'Absolute path or glob pattern for ignoring included files.')
+    .option('-l, --lint', 'Lint the input file and report any errors.')
+    .option('-r, --relaxed', 'Use relaxed JSON format.')
+    .option('-p, --public', 'Only values in #public objects will be extracted.')
+    .option('-o, --extract-option <option>', 'Extract the value of an option.')
+    .option('-j, --option-as-json', 'The extracted value will be extracted as JSON.')
+    .option('-m, --allow-missing-option', 'Do not fail when no such value is present.')
+    .option('--json', 'JSON output.')
+    .option('--allowmissing', 'Allow missing options.')
+    .parse();
+
+const argv = program.opts();
+argv._ = program.args;
 
 // This is needed as optimist will not force exit when --help is
 // passed on it's own, and we want it to.
 if (argv.help || argv._.length < 1) {
-    argp.showHelp();
+    program.help();
```

Comment:
We can update it to `program.outputHelp()` so that it doesn't exit with success. The related test can also be fixed back.

Reply:
Thanks for this. I was not aware about it.